core/src/main/java/org/apache/calcite/sql/SqlForStmt.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Parse tree for a {@code SqlForStmt}.
+ */
+public class SqlForStmt extends SqlIterationStmt {
+  public static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("FOR", SqlKind.FOR_STATEMENT);
+
+  public final SqlIdentifier forLoopVariable;
+  public final SqlIdentifier cursorName;
+  public final SqlNode cursorSpecification;
+
+  /**
+   * Creates a {@code SqlForStmt}.
+   * @param pos         Parser position, must not be null.
+   * @param statements  List of statements to iterate, must not be null.
+   * @param beginLabel  Optional begin label, must match end label if not null.
+   * @param endLabel    Optional end label, must match begin label if not null.
+   * @param forLoopVariable
+   *                    The name of the loop.
```

Comment:
nit: Should this be "loop variable"?

Reply:
Also, should we say that it "must not be null"?

Reply:
Updated in https://github.com/googleinterns/calcite/pull/295/commits/0729b1498ee2b87bc6f1697d05000da7b7fb4228